src/parser/expression.js

```diff
@@ -494,14 +494,29 @@ export default class ExpressionParser extends LValParser {
 
       const node = this.startNodeAt(startPos, startLoc);
       node.callee = base;
+
+      // TODO: Clean up/merge this into `this.state` or a class like acorn's
+      // `DestructuringErrors` alongside refShorthandDefaultPos and
+      // refNeedsArrowPos.
+      const refTrailingCommaPos: Pos = { start: -1 };
+
       node.arguments = this.parseCallExpressionArguments(
         tt.parenR,
         possibleAsync,
+        refTrailingCommaPos,
       );
       this.finishCallExpression(node);
 
       if (possibleAsync && this.shouldParseAsyncArrow()) {
         state.stop = true;
+
+        if (refTrailingCommaPos.start > -1) {
```

Comment:
Why do async and non-async parse their parameters differently?

Reply:
I believe they are parsed differently because non-async arrow function are parsed by converting a parenthesized sequence expression, while async functions by converting a call expression (e.g. `(arrow, or, parenthesized)` and `async(call, to, the = "async", function, or, async, arrow, parameters)`).